daemon/daemon.go

```diff
@@ -836,11 +861,24 @@ func NewDaemon(ctx context.Context, config *config.Config, pluginStore *plugin.S
 	}
 	d.configStore.Store(cfgStore)
 
-	// TEST_INTEGRATION_USE_SNAPSHOTTER is used for integration tests only.
-	if os.Getenv("TEST_INTEGRATION_USE_SNAPSHOTTER") != "" {
-		d.usesSnapshotter = true
-	} else {
-		d.usesSnapshotter = config.Features["containerd-snapshotter"]
+	migrationThreshold := int64(-1)
+	tryGraphDriver := graphdriver.IsRegistered
+	if os.Getenv("TEST_INTEGRATION_USE_GRAPHDRIVER") != "" {
+		tryGraphDriver = func(driver string) bool {
+			if driver == "" || graphdriver.IsRegistered(driver) {
+				return true
+			}
+			log.G(ctx).WithField("driver", driver).Warn("TEST_INTEGRATION_USE_GRAPHDRIVER is set but graphdriver is not registered")
+			return false
+		}
+	}
+	if config.Features["containerd-migration"] {
+		// TODO: Allow setting the threshold
+		migrationThreshold = math.MaxInt64
+		log.G(ctx).WithField("max_size", migrationThreshold).Info("(Experimental) Migration to containerd is enabled, driver will be switched to snapshotter after migration is complete")
+	}
+	if config.Features["containerd-snapshotter"] {
+		log.G(ctx).Warn(`"containerd-snapshotter" is now the default and no longer needed to be set`)
```

Comment:
I think we should also handle a case where both containerd-migration is enabled and containerd snapshotter is explicitly disabled.

It's unclear what the expected behavior is in such case, so we should just error out

```suggestion
		log.G(ctx).Warn(`"containerd-snapshotter" is now the default and no longer needed to be set`)
	}
	if enabled, ok := config.Features["containerd-snapshotter"]; ok && !enabled{
		if config.Features["containerd-migration"] {
			// hard failure
		}
```